packages/flame/lib/src/collisions/broadphase/quadtree/quadtree.dart

```diff
@@ -187,8 +193,9 @@ class QuadTree<T extends Hitbox<T>> {
     final node = _hitboxAtNode.remove(hitbox);
     if (node != null) {
       node.hitboxes.remove(hitbox);
+      // TODO(Spydon): Shouldn't this if be negated?
       if (keepOldPosition) {
-        _oldPositionByItem.remove(node);
+        _oldPositionByItem.remove(hitbox);
```

Comment:
Does this change make sense @ASGAlex? And also, shouldn't `keepOldPosition` be inverted in the `if`?

Reply:
Yes, you are right the both things are bugs. 
But I see that function that uses it - `hasMoved` - is not used anywhere, just mentioned in comments. 
Looks like I thought about future optimization here but idea was lost then...
So we have two options here: 
1. Fixing mentioned bugs
2. Safely remove unsed functionality.

Reply:
Alright, sounds like the best option is removing it then.
Would you like to make a PR for removing it? :)

Reply:
@ASGAlex Hmm, but how can it work if `hasMoved` isn't used? Shouldn't that be used to determine if new quadrants might need to be created, or does it always check that now?